src/Uno.UI/UI/Xaml/Internal/EventManager.cs

```diff
@@ -0,0 +1,159 @@
+﻿#nullable enable
+
+#if UNO_HAS_ENHANCED_LIFECYCLE
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Uno.UI.Dispatching;
+
+namespace Uno.UI.Xaml.Core;
+
+internal sealed class EventManager
+{
+	private List<DependencyObject?>? _loadedEventList;
```

Comment:
There is like 100% chances this list will be created, I would not make it nullable and remove all the `null` checks in the class. Also you should init with with a large number to avoid resizing:

```suggestion
	private List<DependencyObject?> _loadedEventList = new(1024);
```

Note: the comment is probably also relevant for the `_effectiveViewportChangedQueue`